### PR TITLE
Add administrator managment view

### PR DESCRIPTION
**Pull request by ReeceHoffmann**
Changes:
- update UI to use `administrator_role` in place of legacy `administrator`
- add administrator management tab to admin view
- add complete data models for user(s), account, groups, and permissions. Includes all supporting models
- add complete fake data generation of user(s), account, groups, and permissions
- add `GET` api route mocking for `/account`, `/admin/roles`, `/admin/users`, and 
- add api `PUT` route mocking for `admin/users/{user_id}/role` 
- update tests as required

### CODE REVIEW

File: src/js/administration/components/administrators/Administrators.tsx [hunk 1]
```diff
@@ -35,7 +37,7 @@ export const ManageAdministrators = () => {
     return (
         <>
             <Toolbar>
-                <InputSearch name="search" aria-label="search" value={term} onChange={e => setTerm(e.target.value)} />
+                <InputSearch name="search" aria-label="search" value={term} onChange={onSearch} />
```
Comment (igboyes) [inline src/js/administration/components/administrators/Administrators.tsx]: What is the point of this?

File: src/js/administration/components/administrators/Item.tsx [hunk 2]
```diff
@@ -46,10 +46,10 @@ export const AdministratorItem = ({ user, roles }: AdministratorItemProps) => {
 
     return (
         <StyledAdministrator key={user.id}>
-            <InitialIcon handle={user.handle} size={"lg"} />
+            <InitialIcon handle={user.handle} size="lg" />
             <UserHandle>{user.handle}</UserHandle>
             <InlineRoleSelect value={user.administrator_role} roles={roles} onChange={onChange} />
-            <Icon name="trash" color="red" onClick={() => onChange(null)} aria-label={`remove role`} />
+            <Icon name="trash" color="red" onClick={() => onChange(null)} aria-label="remove role" />
```
Comment (igboyes) [inline src/js/administration/components/administrators/Item.tsx]:
```suggestion
            <Icon name="trash" color="red" onClick={() => onChange(null)} aria-label="Remove administrator role" />
```

File: src/js/base/select/SelectButton.tsx [hunk 3]
```diff
@@ -1,17 +1,17 @@
-import * as RadixSelect from "@radix-ui/react-select";
+import { Trigger, Value } from "@radix-ui/react-select";
```
Comment (igboyes) [inline src/js/base/select/SelectButton.tsx]: I prefer that this is unchanged. It makes our code match the examples in the docs and makes it obvious what Radix component `Trigger` and `Value` are associated with.

File: src/js/samples/components/Detail/Rights.jsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ export class SampleRights extends React.Component {
                         </InputGroup>
 
                         <InputGroup>
-                            <InputLabel>All Users' Rights</InputLabel>
+                            <InputLabel>{"All Users' Rights"}</InputLabel>
```
Comment (igboyes) [inline src/js/samples/components/Detail/Rights.jsx]: Why was this necessary?

File: src/js/utils/reducers.js [hunk 5]
```diff
@@ -24,10 +24,22 @@ export const insert = (state, payload, sortKey, sortReverse = false) => {
...
+export const updateMember = (list, payload) => {
+    if (!list) {
+        return list;
+    }
+    return map(list, item => {
+        if (item.id === payload.id) {
+            return payload;
+        }
+        return item;
+    });
```
Comment (igboyes) [inline src/js/utils/reducers.js]: Restructure so condition is `if (list) {}`.

**Review by igboyes — CHANGES_REQUESTED** [review]
Please change combox files names like `SearchBar` -> `ComboBoxSearchBar`. This is a convention we have not been following.

This works well. I'm going to merge and make changes to style later if that's okay with you. There are a few minor accessibility issues and edge cases I will make issues for.

File: src/js/samples/components/Detail/Rights.jsx (hunk 4 above)
Comment (ReeceHoffmann) [reply src/js/samples/components/Detail/Rights.jsx]: it isn't... reverted

**Comment by igboyes** [comment]
Resolve conflicts on this @ReeceHoffmann and I will merge.

**Review by igboyes — APPROVED** [review]

**Pull request merged**